core/src/main/scala/org/bykn/bosatsu/PackageMap.scala

```diff
@@ -313,9 +321,24 @@ object PackageMap {
       Memoize.memoizeDagFuture[ResolvedU, Ior[NonEmptyList[
         PackageError
       ], (TypeEnv[Kind.Arg], Package.Inferred)]] {
-        // TODO, we ignore importMap here, we only check earlier we don't
-        // have duplicate imports
-        case (Package(nm, imports, exports, (stmt, _)), recurse) =>
+        case (Package(nm, imports, exports, (stmt, imps)), recurse) =>
+
+          val nameToRes = imports.iterator.map { i => 
+            val resolved: Package.Resolved = i.pack
+            val name = FixType.unfix(resolved) match {
+              case Left(iface) => iface.name
+              case Right(pack) => pack.name
+            }
+            
+            (name, resolved)
+          }
+          .toMap
+
+          def resolvedImports: ImportMap[Package.Resolved, Unit] = 
+            imps.traverse[cats.Id, Package.Resolved, Unit] { (p, i) =>
+              (nameToRes(p), i)
```

Comment:
This apply should be safe since we have constructed this ImportMap from the same source.

Add a comment with this explanation